sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/TypeCoercion.scala

```diff
@@ -1121,25 +1121,6 @@ object TypeCoercion extends TypeCoercionBase {
       // Hive treats (true = 1) as true and (false = 0) as true,
       // all other cases are considered as false.
 
-      // We may simplify the expression if one side is literal numeric values
-      // TODO: Maybe these rules should go into the optimizer.
-      case EqualTo(bool @ BooleanType(), Literal(value, _: NumericType))
-        if trueValues.contains(value) => bool
-      case EqualTo(bool @ BooleanType(), Literal(value, _: NumericType))
-        if falseValues.contains(value) => Not(bool)
```

Comment:
These transformations apply as optimization to remove `Equal`. After this change, do we still have such optimization as before?

Reply:
Yes, UnwrapCastInBinaryComparison nicely takes care of it.

Reply:
Maybe you can document it here saying those boolean simplifications are removed because UnwrapCastInBinaryComparison can take care of it.